Feature/readme (#67)
iceberg-project/Seals: README.md

[![Build Status](https://travis-ci.com/iceberg-project/Seals.svg?branch=devel)](https://travis-ci.com/iceberg-project/Seals)

## Prerequisites - all available on bridges via the commands below
- Linux
- Python 3
- CPU or NVIDIA GPU + CUDA CuDNN

## Software Dependencies - these will be installed automatically with the installation below.
- numpy
- scipy
- pandas
- torch==0.4.0
- torchvision==0.2.1
- tensorboardX==1.8
- opencv-python
- rasterio
- affine
- geopandas
- pandas

## Installation
Preliminaries:

> Login to Bridges via ssh using a Unix or Mac command line terminal.  Login is available to Bridges directly or through the XSEDE portal. Please see the <a href="https://portal.xsede.org/psc-bridges">Bridges User's Guide</a>.  

Reviewer: ```suggestion
Login to Bridges via ssh using a Unix or Mac command line terminal.  Login is available to Bridges directly or through the XSEDE portal. Please see the [Bridges User's Guide](https://portal.xsede.org/psc-bridges).  
```

For Windows Users:  

> Many tools are available for ssh access to Bridges.  Please see <a href="https://ubuntu.com/tutorials/tutorial-ubuntu-on-windows#1-overview">Ubuntu</a>, <a href="https://mobaxterm.mobatek.net/">MobaXterm</a>, or <a href="https://www.chiark.greenend.org.uk/~sgtatham/putty/">PuTTY</a>

Reviewer: ```suggestion
Many tools are available for ssh access to Bridges.  Please see [Ubuntu](https://ubuntu.com/tutorials/tutorial-ubuntu-on-windows#1-overview), [MobaXterm](https://mobaxterm.mobatek.net) or [PuTTY](https://www.chiark.greenend.org.uk/~sgtatham/putty/)
```

### PSC Bridges
Once you have logged into Bridges, you can follow one of two methods for installing iceberg-seals.

> Method #1 (Recommended):  

Reviewer: ```suggestion
#### Method 1 (Recommended):  
```

```bash
$ pwd
/home/username
$ cd $SCRATCH                      # switch to your working space.
$ mkdir Seals                      # create a directory to work in.
$ cd Seals                         # move into your working directory.
$ module load cuda                 # load parallel computing architecture.
$ module load python3              # load correct python version.
$ virtualenv seals_env             # create a virtual environment to isolate your work from the default system.
$ source seals_env/bin/activate    # activate your environment. Notice the command line prompt changes to show your environment on the next line.
[seals_env] $ pwd
/pylon5/group/username/Seals
[seals_env] $ export PYTHONPATH=<path>/seals_env/lib/python3.5/site-packages # set a system variable to point python to your specific code. (Replace <path> with the results of pwd command above.
[seals_env] $ pip install iceberg_seals.search # pip is a python tool to extract the requested software (iceberg_seals.search in this case) from a repository. (this may take several minutes).
```

> Method #2 (Installing from source; recommended for developers only): 

Reviewer: ```suggestion
#### Method 2 (Installing from source; recommended for developers only): 
```

```bash
$ git clone https://github.com/iceberg-project/Seals.git
$ module load cuda
$ module load python3
$ virtualenv seals_env
$ source seals_env/bin/activate
[seals_env] $ export PYTHONPATH=<path>/seals_env/lib/python3.5/site-packages
[seals_env] $ pip install . --upgrade
```

To test
```bash
[iceberg_seals] $ deactivate    # exit your virtual environment.
$ interact -p GPU-small            # request a compute node (this may take a minute or two or more).
$ cd $SCRATCH/Seals             # make sure you are in the same directory where everything was set up before.
$ module load cuda                 # load parallel computing architecture, as before.
$ module load python3              # load correct python version, as before.
$ source seals_env/bin/activate # activate your environment, no need to create a new environment because the Seals tools are installed and isolated here.
[iceberg_seals] $ iceberg_seals.detect --help  # this will display a help screen of available usage and parameters.
```
## Prediction

> - Download a pre-trained model at: https://github.com/iceberg-project/Seals/tree/feature/README/models/Heatmap-Cnt/UnetCntWRN/UnetCntWRN_ts-vanilla.tar 

Reviewer: We may have to change this path when it is merged.

Author: Changed to master since the model is already there.

[![Build Status](https://travis-ci.com/iceberg-project/Seals.svg?branch=devel)](https://travis-ci.com/iceberg-project/Seals)

## Prerequisites - all available on bridges via the commands below
- Linux
- Python 3
- CPU and NVIDIA GPU + CUDA CuDNN

## Software Dependencies - these will be installed automatically with the installation below.
- numpy
- scipy
- pandas
- torch==0.4.0
- torchvision==0.2.1
- tensorboardX==1.8
- opencv-python
- rasterio
- affine
- geopandas
- pandas

## Installation
Preliminaries:  

> These instructions are specific to XSEDE/bridges but other resources can be used if cuda, python3, and a gpu processor are present, in which case 'module load' instructions can be skipped, which are specific to bridges.  

Reviewer: ```suggestion
These instructions are specific to XSEDE Bridges but other resources can be used if cuda, python3, and a NVIDIA P100 GPU are present, in which case 'module load' instructions can be skipped, which are specific to Bridges.  
```
